be/src/storage/cumulative_compaction.cpp

```diff
@@ -65,62 +81,87 @@ Status CumulativeCompaction::pick_rowsets_to_compact() {
 
     std::sort(candidate_rowsets.begin(), candidate_rowsets.end(), Rowset::comparator);
 
-    RETURN_IF_ERROR(check_version_continuity_with_cumulative_point(candidate_rowsets));
-    RETURN_IF_ERROR(check_version_continuity(candidate_rowsets));
-
     std::vector<RowsetSharedPtr> transient_rowsets;
     size_t compaction_score = 0;
     // the last delete version we meet when traversing candidate_rowsets
     Version last_delete_version{-1, -1};
 
+    int64_t prev_end_version = _tablet->cumulative_layer_point() - 1;
     for (auto rowset : candidate_rowsets) {
+        // meet missed version
+        if (rowset->start_version() != prev_end_version + 1) {
+            if (!transient_rowsets.empty() &&
+                compaction_score >= config::min_cumulative_compaction_num_singleton_deltas) {
+                // min_cumulative_compaction_num_singleton_deltas = 2
+                // 7,6,<4,3>|2-1 -> <4,3>
+                _input_rowsets = transient_rowsets;
+                break;
+            } else {
+                // 7,6,<3>|2-1
```

Comment:
1, 2, 3, 6, 7
The cumulative point is 3, and the missed versions are 4 and 5.

Reply:
version 3 won't to do cumulative compaction is by design since we want to avoid fragmented compaction.